options/ansi/generic/time-stubs.cpp

```diff
@@ -578,10 +579,445 @@ char *ctime_r(const time_t *clock, char *buf) {
 	return asctime_r(localtime(clock), buf);
 }
 
-char *strptime(const char *__restrict, const char *__restrict,
+namespace {
+
+int read_number(int *result, size_t *string_index, const char *s, int ndigits) {
```

Comment:
How is this different than a ``scanf()``?

Reply:
`sscanf()` doesn't return the number of characters read, but the number of items. I need the number of characters read to keep track of the index in the input string and so on. I could re-implement `read_number()` using `sscanf()` but I'm not sure if this is necessary.

Reply:
You can use scanf's ``%n`` for this.